File: backend/springboot/src/main/java/com/technight/api/dto/UpdateExampleDto.java
package com.technight.api.dto;

import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * DTO for updating an existing Example
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
@Schema(description = "DTO for updating an existing Example")
public class UpdateExampleDto {
    
    /**
     * Name of the example
     */
    @Size(max = 200, message = "Name must not exceed 200 characters")
    @Schema(description = "Name of the example", example = "Updated Example")
    private String name;
    
    /**
     * Title of the example
     */
    @Size(max = 200, message = "Title must not exceed 200 characters")
    @Schema(description = "Title of the example", example = "Updated Title")
    private String title;
    
    /**
     * Optional description
     */
    @Size(max = 1000, message = "Description must not exceed 1000 characters")
    @Schema(description = "Optional description", example = "This is an updated description")
    private String description;
    
    /**
     * Indicates if the example is active
     */
    @Schema(description = "Indicates if the example is active", example = "false")
    private Boolean isActive;
}

File: backend/dotnet/TechnightApi/DTOs/UpdateExampleDto.cs
using System.ComponentModel.DataAnnotations;

namespace TechnightApi.DTOs;

/// <summary>
/// DTO for updating an existing Example
/// </summary>
public class UpdateExampleDto
{
    /// <summary>
    /// Name of the example
    /// </summary>
    [StringLength(200)]
    public string? Name { get; set; }

    /// <summary>
    /// Title of the example
    /// </summary>
    [StringLength(200)]
    public string? Title { get; set; }

    /// <summary>
    /// Optional description
    /// </summary>
    [StringLength(1000)]
    public string? Description { get; set; }

    /// <summary>
    /// Indicates if the example is active
    /// </summary>
    public bool? IsActive { get; set; }
}

